src/main/java/com/digitalpebble/storm/crawler/bolt/ParserBolt.java

```diff
@@ -300,8 +305,17 @@ public void execute(Tuple tuple) {
             }
         }
 
-        collector.emit(tuple, new Values(url, content, metadata, text.trim(),
-                slinks));
+        for (String outlink : slinks) {
+            // configure which metadata gets inherited from parent
+            Map<String, String[]> linkMetadata = metadataTransfer
+                    .getMetaForOutlink(metadata);
+            collector
+                    .emit(com.digitalpebble.storm.crawler.Constants.StatusStreamName,
+                            tuple, new Values(outlink, linkMetadata,
```

Comment:
I'm still not convinced that this is the best way of categorizing the content but that's another matter (e.g. can't use sitemaps for discovery then). It makes sense to be able to track the path indeed, at least for debugging. 

Could do that by modifying [MetadataTransfer](https://github.com/DigitalPebble/storm-crawler/blob/master/core/src/main/java/com/digitalpebble/storm/crawler/util/MetadataTransfer.java#L50) so that it takes the sourceURL as argument and have the option to add that to a metadatum. It would inherit the ones from the parent metadata.